src/util/std_expr.cpp

```diff
@@ -134,3 +136,136 @@ void let_exprt::validate(const exprt &expr, const validation_modet vm)
       "let bindings must be type consistent");
   }
 }
+
+static optionalt<exprt> substitute_symbols_rec(
+  const std::map<irep_idt, exprt> &substitutions,
+  exprt src)
```

Comment:
Is it necessary to pass by value here (a copy is taken below)?

Reply:
The copy happens in most cases -- so this is the pattern "use pass by value when copy is inevitable".